Charades_Game_In_JavaScript/Charades Game In JavaScript final blog.md

```diff
@@ -0,0 +1,348 @@
+Inspired by Marina's post, I included the "hide" and "show" properties in the CSS file.
+
+##### **in CSS:**
+You could set the style of the timer by its tag name, and make sure you include `display: none;` so that it is not shown in the beginning.
+The `none` indicates it is not showing, and `block` indicates it is showing.
+
+![](https://i.imgur.com/2rYdVql.png)
+
+
+##### **in JavaScript**
+When the button is clicked, we want the buttons and the texts disappear, while the countdown timer should show up. Therefore, **inside each function**, we would use `.classList.toggle` to hide texts, and change button style to `"hidden"` to hide buttons.
+
+Here we take `time()` for example, and the `word()` is implemented in the exact same way.
+
+1. Hide Texts:
+To select the text boxes, we set it to a variable `var element = document.getElementById("intro-words");`. Then, we would modify the style of this element into `"hide"`. Same for the `"intro-time"` text box.
+
+![](https://i.imgur.com/0NXOPNd.png)
+
+2. Hide Buttons:
+Since there are two buttons under the `but` class, we need to select them with `const button = document.querySelectorAll('.but');`, then access each of them by using `forEach(but => but...);` The button have a style property called visibility, so we could simple make it to "hidden".
+
+![](https://i.imgur.com/VNnoB6d.png)
+
+3. Show the Counter:
+Same as hiding the text boxes, we would select the timer element by its id `"go"`, and use `.classList.toggle` to set it to `"show"`.
+
+
+### **Phase 2: Timer**
+Now we could implement the timer.
+The idea is that the timer will take in the `textContent` of our `"go"` element, and show the modified `textContent` every one second. And we update the `textContent` by keeping a counter and decrement it every second.
+
+##### **in JavaScript:**
+We first create a variable `seconds` to hold the value in the `textContent` of our `"go"` element, which is 5.
+
+![](https://i.imgur.com/itWasch.png)
+
+Then, we would use `setInterval()` function to let the timer count down at each second (=1000 milliseconds), and update the variable `seconds` by using the decrementing operator `seconds--;`, and also, update the `textContent` with the decremented seconds or whatever words you want (like "Ready??" "Go!!").
+
+We use an `if-else if-else` loop to separate conditions on the seconds. The seconds would go into the `else` statement after 5 iterations, and we would `clearInterval(count);` and jump to the corresponding page.
+
+![](https://i.imgur.com/k6gd2dv.png)
+
+Now try out the new feature you got for your limiting time version! 
+
+### **Phase 3: Copy to Both Functions**
+
+##### **in JavaScript:**
+Simply copy and paste all the codes we added in the function `time()` in phase 1 and 2 into the function `word()`. Remember to change the redirecting page into `"card_word.html"`!
```

Comment:
clarify this when you clarify line 172 to make sure the explanations are consistent